pkg/lang/csharp/plugin_expose.go

```diff
@@ -0,0 +1,670 @@
+		}
+
+		for _, prefix := range routePrefixes {
+			spec := actionSpec{
+				name:             method.Name,
+				method:           method,
+				routeTemplate:    path.Join(prefix, routeTemplate),
+				verb:             verb,
+				hasRouteTemplate: len(args) > 0,
+			}
+			specs = append(specs, spec)
+		}
+	}
+
+	for _, acceptAttr := range acceptVerbsAttrs {
+		var verbs []core.Verb
+		args := acceptAttr.Args()
+		routeTemplate := ""
+		hasRouteTemplate := false
+		for _, arg := range args {
+			if arg.Name == "" {
+				verb := core.Verb(strings.ToUpper(arg.Value))
+				_, supported := core.Verbs[verb]
+
+				if supported {
+					verbs = append(verbs, verb)
+				}
+			}
+			if arg.Name == "Route" {
+				routeTemplate = arg.Value
+				hasRouteTemplate = true
+			}
+		}
+
+		for _, verb := range verbs {
+			for _, prefix := range routePrefixes {
+				spec := actionSpec{
+					verb:             verb,
+					name:             method.Name,
+					method:           method,
+					routeTemplate:    path.Join(prefix, routeTemplate),
+					hasRouteTemplate: hasRouteTemplate,
+				}
+				specs = append(specs, spec)
+			}
+		}
+	}
+	return specs
+}
+
+// sanitizeConventionalPath converts ASP.net conventional path parameters to Express syntax,
+// but does not perform validation to ensure that the supplied string is a valid ASP.net route.
+// As such, there's no expectation of correct output for invalid paths
+// Regexp constraints and controller/action token replacement are not yet supported
+func sanitizeConventionalPath(path string) string {
+	// convert to longest possible proxy route when required
+	firstProxyParamIndex := findFirstProxyRouteIndicator(path)
+	if firstProxyParamIndex > -1 {
+		path = path[0:firstProxyParamIndex]
+		path = path[0:strings.LastIndex(path, "{")+1] + "rest*}"
+	}
+
+	// convert path params to express syntax
+	path = regexp.MustCompile("{([^:}]*):?[^}]*}").ReplaceAllString(path, ":$1")
+	return path
+}
+
+func sanitizeAttributeBasedPath(path string, area string, controller string, action string) string {
+	//TODO: handle regex constraints -- they may include additional curly braces ("{", "}") that aren't currently accounted for
+
+	// replace params such as {controller=Index}
+	specialParamFormat := `(?i){\s*%s\s*=\s*%s\s*}`
+	path = regexp.MustCompile(fmt.Sprintf(specialParamFormat, "area", area)).ReplaceAllString(path, area)
+	path = regexp.MustCompile(fmt.Sprintf(specialParamFormat, "controller", controller)).ReplaceAllString(path, controller)
+	path = regexp.MustCompile(fmt.Sprintf(specialParamFormat, "action", action)).ReplaceAllString(path, action)
+
+	// convert to longest possible proxy route when required
+	firstProxyParamIndex := findFirstProxyRouteIndicator(path)
+	if firstProxyParamIndex > -1 {
+		path = path[0:firstProxyParamIndex]
+		path = path[0:strings.LastIndex(path, "{")+1] + "rest*}"
+	}
+
+	// convert path params to express syntax
+	path = regexp.MustCompile("{([^:}]*):?[^}]*}").ReplaceAllString(path, ":$1")
```

Comment:
Does the `{`/`}` need to be escaped?

Reply:
Not as far as I'm aware. 

This replacement pattern replaces the following examples:
`{myParam}` ,`{myParam:int}` 
with:
`:myParam`

Reply:
Hm, usually I see `{` causing parsing errors because of the regexp meaning of `{}` for repetitions. If it doesn't then that's fine - I guess since it can tell since there's no numbers or `,` in there that it's unambiguous in matching the characters instead of being a control character.

Reply:
Ah, that hadn't even occurred to me for some reason! Yeah, it must recognize that we're not defining repetitions here, but I can definitely escape `{` to eliminate that risk.